src/rest_offsets.lua

```diff
@@ -64,34 +71,51 @@ function get_user_choices()
 
     dialog:CreateOkButton()
     dialog:CreateCancelButton()
-    return (dialog:ExecuteModal(nil) == finale.EXECMODAL_OK), answer[1]:GetInteger(), answer[2]:GetInteger()
+    dialog:RegisterHandleOkButtonPressed(function()
+        config.offset = answer[1]:GetInteger()
+        config.layer = answer[2]:GetInteger()
+    end)
+    dialog:RegisterCloseWindow(function()
+        dialog:StorePosition()
+        config.pos_x = dialog.StoredX
+        config.pos_y = dialog.StoredY
+    end)
+    return dialog
 end
 
-function change_rest_offset()
-    local base_offset = 6 -- default rest ZERO vertical position
-    local is_ok = false -- (rest_offset and layer_number are global vars)
-    is_ok, rest_offset, layer_number = get_user_choices()
-    if not is_ok then
-        return
-    end -- user cancelled
-
-    if rest_offset < -60 or rest_offset > 60 then
-        show_error("bad_offset", rest_offset)
-        return
-    end
-    if layer_number < 0 or layer_number > 4 then
-        show_error("bad_layer_number", layer_number)
-        return
-    end
+function make_the_change()
     if finenv.RetainLuaState ~= nil then
         finenv.RetainLuaState = true
     end
-
-    for entry in eachentrysaved(finenv.Region(), layer_number) do
-        if entry:IsRest() then
-            entry:SetRestDisplacement(rest_offset + base_offset)
+    for entry in eachentrysaved(finenv.Region(), config.layer) do
+         if entry:IsRest() then
+            if config.offset == 0 then
+                entry:SetFloatingRest(true)
+            else
+                entry:MakeMovableRest()
+                local rest = entry:GetItemAt(0)
+                local curr_staffpos = rest:CalcStaffPosition()
+                entry:SetRestDisplacement(entry:GetRestDisplacement() + config.offset - curr_staffpos - 4)
```

Comment:
Instead of hard-coding `-4`, you should be using the correct rest position for the rest from `FCCurrentStaffSpec`. The properties are `DoubleWholeRestPosition`, `WholeRestPosition`, `HalfRestPosition`, and `OtherRestPosition`.

Also, it might be nice to add this to the `note_entry` library as a callable function.

Reply:
> might be nice to add this to the `note_entry` library as a callable function

True. I'll work this up in due course to match rest duration to StaffSpec. Not confident enough to start editing a library file! For now this script works miles better than the old version currently in the repo.